test/cli/cli_test.go

```diff
@@ -2,308 +2,59 @@
-		binaryPath += `.exe`
-	}
-
-	// Envs
-	envs, err := env.FromOs()
-	assert.NoError(t, err)
-	envs.Set("TARGET_PATH", binaryPath)
-	envs.Set("SKIP_API_CODE_REGENERATION", "1")
-
-	// Build binary
-	var stdout, stderr bytes.Buffer
-	cmd := exec.Command("make", "build-local")
-	cmd.Dir = projectDir
-	cmd.Env = envs.ToSlice()
-	cmd.Stdout = &stdout
-	cmd.Stderr = &stderr
-	if err := cmd.Run(); err != nil {
-		t.Fatalf("Compilation failed: %s\n%s\n%s\n", err, stdout.Bytes(), stderr.Bytes())
-	}
-
-	return binaryPath
-}
-
-// AssertExpectations compares expectations with the actual state.
-func AssertExpectations(
-	t *testing.T,
-	envProvider testhelper.EnvProvider,
-	testDirFs filesystem.Fs,
-	workingDirFs filesystem.Fs,
-	exitCode int,
-	inOut *cmdInputOutput,
-	project *testproject.Project,
-) {
-	t.Helper()
-
-	// Compare stdout
-	expectedStdoutFile, err := testDirFs.ReadFile(filesystem.NewFileDef("expected-stdout"))
-	assert.NoError(t, err)
-	expectedStdout := testhelper.MustReplaceEnvsString(expectedStdoutFile.Content, envProvider)
-
-	// Compare stderr
-	expectedStderrFile, err := testDirFs.ReadFile(filesystem.NewFileDef("expected-stderr"))
-	assert.NoError(t, err)
-	expectedStderr := testhelper.MustReplaceEnvsString(expectedStderrFile.Content, envProvider)
-
-	// Get outputs
-	stdout := inOut.StdoutString()
-	stderr := inOut.StderrString()
-
-	// Compare exit code
-	expectedCodeFile, err := testDirFs.ReadFile(filesystem.NewFileDef("expected-code"))
-	assert.NoError(t, err)
-	expectedCode := cast.ToInt(strings.TrimSpace(expectedCodeFile.Content))
-	assert.Equal(
-		t,
-		expectedCode,
-		exitCode,
-		"Unexpected exit code.\nSTDOUT:\n%s\n\nSTDERR:\n%s\n\n",
-		stdout,
-		stderr,
+	testsDir := filepath.Dir(testFile)
+	rootDir := filepath.Join(testsDir, "..", "..")
+
+	r := runner.NewRunner(t, testsDir)
+	binaryPath := r.CompileBinary(
+		rootDir,
+		"bin_func_tests",
+		"TARGET_PATH",
+		"build-local",
 	)
 
-	// Assert STDOUT and STDERR
-	wildcards.Assert(t, expectedStdout, stdout, "Unexpected STDOUT.")
-	wildcards.Assert(t, expectedStderr, stderr, "Unexpected STDERR.")
-
-	// Expected state dir
-	expectedDir := "out"
-	if !testDirFs.IsDir(expectedDir) {
-		t.Fatalf(`Missing directory "%s" in "%s".`, expectedDir, testDirFs.BasePath())
-	}
-
-	// Copy expected state and replace ENVs
-	expectedDirFs := aferofs.NewMemoryFsFrom(filesystem.Join(testDirFs.BasePath(), expectedDir))
-	testhelper.MustReplaceEnvsDir(expectedDirFs, `/`, envProvider)
-
-	// Compare actual and expected dirs
-	testhelper.AssertDirectoryContentsSame(t, expectedDirFs, `/`, workingDirFs, `/`)
-
-	// Check project state
-	expectedStatePath := "expected-state.json"
-	if testDirFs.IsFile(expectedStatePath) {
-		// Read expected state
-		expectedSnapshot, err := testDirFs.ReadFile(filesystem.NewFileDef(expectedStatePath))
+	loadArgsFn := func(t *runner.Test) []string {
+		// Load command arguments from file
+		argsFile, err := t.TestDirFS().ReadFile(filesystem.NewFileDef("args"))
```

Comment:
Toto by teoreticky mohla byt priamo sucast `runner`, asi nie je dovod to mat nejak custom. Detail.

Reply:
Tj. `WithRunCLIBinary` metoda by mala iba jeden argument.

Reply:
A jo, já to měl původně jako samostatnou metodu (co se zapínala přes `test.Run(WithLoadArgs)`) ale pak mi to přišlo příliš CLI specific, ale je pravda že asi nehrozí že by si jiný CLI testy (co by volaly taky `test.runCLIBinary()`) načítaly argumenty jinak.